cosmos/lib/cosmos/utilities/s3_file_cache.rb

```diff
@@ -165,8 +167,11 @@ def initialize(name = 'default', max_disk_usage = MAX_DISK_USAGE)
     end
 
     # Create local file cache location
-    @cache_dir = File.join(Dir.tmpdir, 'cosmos', 'file_cache', name)
+    @cache_dir = Dir.mktmpdir
     FileUtils.mkdir_p(@cache_dir)
+    at_exit do
+      FileUtils.remove_dir(@cache_dir, true)
+    end
 
     # Clear out local file cache
     FileUtils.rm_f Dir.glob("#{@cache_dir}/*")
```

Comment:
Probably don't need to do this if you're dynamically creating a new dir above

Reply:
good catcg